mlir/include/mlir/Interfaces/SideEffectInterfaces.td

```diff
@@ -136,6 +141,10 @@ def ConditionallySpeculatable : OpInterface<"ConditionallySpeculatable"> {
 def AlwaysSpeculatable : TraitList<[
     ConditionallySpeculatable, AlwaysSpeculatableImplTrait]>;
 
+// Marks an Operation as never speculatable.
+def NeverSpeculatable : TraitList<[
+    ConditionallySpeculatable, NeverSpeculatableImplTrait]>;
```

Comment:
I don't follow, why would an op implement `ConditionallySpeculatable` when the answer is never?

This should be the same behavior as doing nothing.

Reply:
Ah right, if an op doesn't implement `ConditionallySpeculatable` then it's just assumed to never be speculatable. 

Thanks, I'll close this PR then.